goose.go

```diff
@@ -1937,8 +1958,7 @@ func (ctx Ctx) returnType(results *ast.FieldList) coq.Type {
 	var ts []coq.Type
 	for _, r := range rs {
 		if len(r.Names) > 0 {
-			ctx.unsupported(r, "named returned value")
-			return coq.TypeIdent("<invalid>")
+			ts = append(ts, ctx.coqType(r.Type.(*ast.FuncType).Results.List[0].Type))
```

Comment:
This is still incorrectly allowing named return values. It looks like you might be expecting `results` to be a function type; in that situation you'll need to call this function on `r.Type.(*ast.FuncType).Results` or something similar, rather than modifying `returnType` itself (which is only intended to process a return type declaration).

Reply:
I ended up reverting this, thanks!